### PR TITLE
[tests-only][full-ci] Json drill for `coreApiCapabilities` Suite using library

### PR DESCRIPTION
**Pull request by SagarGi**
### Description
This is a sample PR to drill down the json response using library `helmich/phpunit-json-assert`.

### Note
Other API `suites` be followed up with PR by PR similar to this.

### Related Issue:
https://github.com/owncloud/ocis/issues/3179

### CODE REVIEW

File: tests/acceptance/features/bootstrap/AppConfigurationContext.php [hunk 1]
```diff
@@ -132,7 +132,21 @@ public function getAppParameter(string $capabilitiesApp, string $capabilitiesPat
...
-	public function userGetsCapabilities(string $username):void {
+	public function userRetrievesCapabilities(string $username): void {
+		$user = $this->featureContext->getActualUsername($username);
+		$this->userGetsCapabilities($user, true);
+	}
+
+	/**
+	 *
+	 * @param string $username
+	 * @param boolean $formatJson // this  parameter if true formats the response in json
```
Comment (grgprarup) [inline tests/acceptance/features/bootstrap/AppConfigurationContext.php:143]:
```suggestion
	 * @param boolean $formatJson // if true then formats the response in json
```

File: tests/acceptance/features/bootstrap/AppConfigurationContext.php [hunk 2]
```diff
@@ -141,7 +155,7 @@ public function userGetsCapabilities(string $username):void {
 				$user,
 				$password,
 				'GET',
-				'/cloud/capabilities',
+				($formatJson) ? '/cloud/capabilities?format=json' : '/cloud/capabilities',
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/AppConfigurationContext.php:158]:
```suggestion
				'/cloud/capabilities' . ($formatJson ? '?format=json' : ''),
```

File: tests/acceptance/features/bootstrap/AppConfigurationContext.php [hunk 3]
```diff
@@ -158,7 +172,7 @@ public function userGetsCapabilities(string $username):void {
 	 * @throws Exception
 	 */
 	public function userGetsCapabilitiesCheckResponse(string $username):void {
-		$this->userGetsCapabilities($username);
+		$this->userGetsCapabilities($username, false);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/AppConfigurationContext.php:175]: default value
```suggestion
		$this->userGetsCapabilities($username);
```

File: tests/acceptance/features/bootstrap/CapabilitiesContext.php [hunk 4]
```diff
@@ -39,81 +39,16 @@ class CapabilitiesContext implements Context {
...
-		);
-	}
-
 	/**
 	 * @Then the major-minor-micro version data in the response should match the version string
 	 *
 	 * @return void
 	 * @throws Exception
 	 */
 	public function checkVersionMajorMinorMicroResponse():void {
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/CapabilitiesContext.php:48]: isn't that checked by 
```
                    "versionstring": {
                      "type": "string",
                      "enum": ["%versionstring%"]
                    }
```
I don't understand why we even have that step, I think the expectation AND the result comes from the same response

Comment (SagarGi) [reply tests/acceptance/features/bootstrap/CapabilitiesContext.php:48]: we are actually asserting information from this `version` with  `versionstring` from the same response
```
"version": {
                "major": 10,
                "minor": 11,
                "micro": 0,
                "string": "10.11.0",
                "edition": "Community",
                "product": "Infinite Scale",
                "productversion": "2.0.0+22f3577b7"
            } ```

**Review by individual-it — APPROVED** [review]

**Review by phil-davis — APPROVED** [review]

**Pull request merged**